Implements DebugDisplay classes.

OmniBlade opened the pull request.

tomsons26 committed:
Implements DebugDisplay classes.

tomsons26 commented on src/game/client/w3ddebugdisplay.cpp:
```diff
@@ -0,0 +1,60 @@
+#include "displaystring.h"
+
+// temp location
+int GameMakeColor(unsigned char red, unsigned char green, unsigned char blue, unsigned char alpha)
```
a home for this needs to be figured out

OmniBlade replied:
I believe an equivalent function already exists in colorspace.h but I didn't know the name for it so just made one up.

tomsons26 commented on src/game/client/w3ddebugdisplay.cpp:
```diff
@@ -0,0 +1,60 @@
+        unicode.Translate(Utf8String(text));
+
+        m_displayString->Set_Text(unicode);
+        m_displayString->Draw(
```
wonky formatting?

tomsons26 commented on src/game/client/debugdisplay.cpp:
```diff
@@ -0,0 +1,59 @@
+{
+    static char _text[5120];
+
+    va_list va;
```
couldn't figure out how to clean this function up

tomsons26 commented on src/game/client/w3ddebugdisplay.h:
```diff
@@ -0,0 +1,41 @@
+    int m_fontWidth;
+    int m_fontHeight;
+    DisplayString *m_displayString;
+};
```
missing new line

tomsons26 commented on src/game/client/debugdisplay.h:
```diff
@@ -0,0 +1,85 @@
+    int m_height;
+    int m_rightMargin;
+    int m_leftMargin;
+};
```
missing new line

tomsons26 commented on src/game/client/debugdisplay.cpp:
```diff
@@ -0,0 +1,59 @@
+            m_xPos += v4;
+        }
+    }
+}
```
missing new line

duncanspumpkin commented:
Maybe it would be an idea to get clang format going as a ci then it means you don't need to comment on the formatting as much.

OmniBlade commented:
Its on my todo list along with moving to github workflows.

OmniBlade committed:
Renames Pack_RGBA to Make_Color.

Fits better with the convention of other functions referring
to color packed to int as just "Color".
